service/history/execution/state_rebuilder.go

```diff
@@ -152,9 +152,6 @@ func (r *stateRebuilderImpl) Rebuild(
 		}
 	}
 
-	if err := rebuiltMutableState.SetCurrentBranchToken(targetBranchToken); err != nil {
```

Comment:
I am not super familiar with this part but it seems that we are validating some things in the version history below. Are we sure the branch token (which is part of version history) being unset doesn't change that logic?